src/main/scala/beam/router/RoutingStatistic.scala

```diff
@@ -0,0 +1,65 @@
+package beam.router
+
+import akka.actor.{Actor, Props}
+import beam.router.BeamRouter.{IterationEndsMessage, IterationStartsMessage, RoutingFailure, RoutingResponse}
+import beam.router.Modes.BeamMode._
+import beam.utils.csv.CsvWriter
+import org.matsim.core.controler.OutputDirectoryHierarchy
+
+/**
+  * @author Dmitry Openkov
+  */
+class RoutingStatistic(ioController: OutputDirectoryHierarchy) extends Actor {
+
+  private val csvHeaders =
+    Seq("departure_time", "origin_x", "origin_y", "destination_x", "destination_y", "mode", "note")
+
+  override def receive: Receive = { case IterationStartsMessage(iteration) =>
+    val writer = new CsvWriter(ioController.getIterationFilename(iteration, "not_found_routes.csv.gz"), csvHeaders)
+    context.become(handleRouterResponses(writer))
+  }
+
+  def handleRouterResponses(writer: CsvWriter): Receive = {
+    case _: IterationEndsMessage =>
+      writer.flush()
+      writer.close()
```

Comment:
will this work correctly for several iterations?

Reply:
Do you mean that the last iteration might not be saved completely?

Reply:
I'll check it and create a new branch in case of issues, because I accidentally merged this one.